saleor/graphql/order/types.py

```diff
@@ -1042,39 +1042,25 @@ def _resolve_user(user):
         return UserByUserIdLoader(info.context).load(root.user_id).then(_resolve_user)
 
     @staticmethod
-    def resolve_shipping_method(root: models.Order, info):
-        external_app_shipping_id = get_external_shipping_id(root)
-
-        if external_app_shipping_id:
-            shipping_method = info.context.plugins.get_shipping_method(
-                checkout=root,
-                channel_slug=root.channel.slug,
-                shipping_method_id=external_app_shipping_id,
-            )
-            if shipping_method:
-                return ChannelContext(
-                    node=shipping_method, channel_slug=root.channel.slug
-                )
+    def resolve_shipping_method(root: models.Order, _info):
+        order_shipping_method_id = root.shipping_method_id or get_external_shipping_id(
+            root
+        )
 
-        if not root.shipping_method_id:
+        if not order_shipping_method_id:
             return None
 
-        def wrap_shipping_method_with_channel_context(data):
-            shipping_method, channel = data
-            return ChannelContext(node=shipping_method, channel_slug=channel.slug)
-
-        shipping_method = ShippingMethodByIdLoader(info.context).load(
-            root.shipping_method_id
+        method = ShippingMethodData(
```

Comment:
we should use the previous way of fetching shipping method here. In case when we use `external_shipping` then we should dynamically generate ShippingMethodData, but only for external shipping.